src/components/ViewClub.js

```diff
@@ -0,0 +1,361 @@
+            });
+
+        // Check if the user has permission to edit this club
+        let userId;
+        if (this.props.client.token) {
+            userId = this.props.client.userIdFromToken();
+        }
+        this.props.client.getMemberships(this.props.name, userId).then(response => {
+            if (response.ok) {
+                response.json().then(body => {
+                    // TODO: Check the user's role
+                    this.setState({
+                        isEditable: true,
+                        memberships: body,
+                    });
+                });
+            } else if (response.status !== 404) {
+                this.setState({
+                    msg: UNEXPECTED_ERROR,
+                    isSuccess: false
+                });
+            }
+        }).catch(() => {
+            this.setState({
+                msg: UNEXPECTED_ERROR,
+                isSuccess: false
+            });
+        });
+    }
+
+    /**
+     * Handles the event that occurs when a user hits submit on the 'Edit Club'
+     * form by attempting to update the club with the information provided.
+     * @param {Event} event
+     */
+    handleSubmit(event) {
+        event.preventDefault();
+        this.props.client.updateClub(
+            this.props.name,
+            this.state.name,
+            this.state.description,
+            this.state.websiteUrl,
+            this.state.facebookUrl,
+            this.state.instagramUrl,
+            this.state.twitterUrl
+        ).then(response => {
+            if (response.ok) {
+                // Display success message
+                this.setState({
+                    msg: CLUB_UPDATED,
+                    isSuccess: true
+                });
+            } else if (response.status === 409) {
+                // A club with that name already exists
+                this.setState({
+                    msg: CLUB_ALREADY_EXISTS,
+                    isSuccess: false
+                });
+            } else if (response.status === 400) {
+                // Some of the information entered is invalid
+                this.setState({
+                    msg: INVALID_INFO,
+                    isSuccess: false,
+                });
+            } else {
+                // An unexpected error occurred
+                this.setState({
+                    msg: UNEXPECTED_ERROR,
+                    isSuccess: false
+                });
+            }
+        });
+    }
+
+    /**
+     * Updates the component state when the user types in an input field.
+     * @param {Event} event
+     */
+    handleInput(event) {
+        event.preventDefault();
+        this.setState({
+            [event.target.name]: event.target.value,
+        });
+    }
+
+    /**
+     * Returns HTML elements displaying the club's info as a form.
+     */
+    renderAsForm() {
+        return (
+            <form onSubmit={this.handleSubmit}>
+                <FormGroup>
```

Comment:
all this copy pasting seems unncessary. Just do something like 
```
let clubInfoUrl = [{ label: 'Name'}, { label: 'Description' }, ... ];
for (let x in clubInfoUrl) {
  return (
               <FormGroup>
                    <Label>{x.label}</Label>
```

Reply:
Umm I don't think that's how return statements work.

Reply:
see https://stackoverflow.com/questions/22876978/loop-inside-react-jsx